api/modules/users/views.py

```diff
@@ -239,6 +240,29 @@ def remove_user_status(request):
     return Response(status=status.HTTP_200_OK)
 
 
+@api_view(['GET'])
+def delete_profile(request):
+    """
+    Remove user profile
+    :param request:
+    :return: 404 if profile does not exist
+    :return: 200 successful
+    """
+    try:
+        user_profile = request.user.profile
+        if user_profile:
+            user_profile.delete()
+        for trip in Trip.objects.all():
```

Comment:
If you have the user object, can't you iterate over just those trips.
Because in current implementation, if you have a total 1000 trips in database and user is in just 2, you will iterate over 1000 trips.

Use `request.user` to get a list of all the trips for that particular user